src/components/sbb-toggle-check/sbb-toggle-check.tsx

```diff
@@ -0,0 +1,101 @@
+import { Component, Prop, h, Event, EventEmitter, Watch } from '@stencil/core';
+import { InterfaceCheckToggleAttributes } from './sbb-link.custom';
+
+let nextId = 0;
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-toggle-check.scss',
+  tag: 'sbb-toggle-check',
+})
+export class SbbToggleCheck {
+  private _checkbox: HTMLInputElement;
+
+  /** Whether the toggle-check is checked. */
+  @Prop({ mutable: true }) public checked = false;
+
+  /** Value of toggle-check. */
+  @Prop() public value?: string;
+
+  /** Name of the toggle-check. */
+  @Prop() public name?: string;
+
+  /** Id of the internal input element - default id will be set automatically. */
+  @Prop() public inputId = `sbb-checkbox-${++nextId}`;
+
+  /** The svg name for the true state - default -> 'tick-small' */
+  @Prop() public icon = 'tick-small';
+
+  /** The disabled prop for the disabled state. */
+  @Prop() public disabled!: boolean;
+
+  /** The required prop for the required state. */
+  @Prop() public required?: boolean;
+
+  /** The label position relative to the toggle. Defaults to 'after' */
+  @Prop() public labelPosition?: InterfaceCheckToggleAttributes['labelPosition'] = 'after';
+
+  /** The aria-label prop for the hidden input. */
+  @Prop() public accessibilityLabel?: string;
+
+  /** The aria-labelledby prop for the hidden input. */
+  @Prop() public accessibilityLabelledby?: string;
+
+  /** The aria-describedby prop for the hidden input. */
+  @Prop() public accessibilityDescribedby?: string;
+
+  /** Event for emiting whenever selection is changed. */
+  @Event() public sbbChange: EventEmitter;
+
+  /** Set checked to the state of the input-checkbox. */
+  private _toggle(): void {
```

Comment:
Late stage nitpick: Would it make sense to inline this functionality since this method is only called from one place and it is a one-liner? (Also toggle is not quite the correct description for what it does.)